### PR TITLE
concreteTest

### PR DESCRIPTION
**Pull request by Pakeesan**

### CODE REVIEW

File: tokyo-supermix-server/src/main/java/com/tokyo/supermix/server/controller/ConcreteTestController.java [hunk 1]
```diff
@@ -1,10 +1,103 @@
...
+            mapper.map(concreteTestList, ConcreteTestResponseDto.class), RestApiResponseStatus.OK),
+        HttpStatus.OK);
+  }
+
+  // get ConcreteTest by id
+  @GetMapping(value = EndpointURI.GET_CONCRETE_TEST_BY_ID)
+  public ResponseEntity<Object> getConcreteTestById(@PathVariable Long id) {
+    if (concreteTestService.isConcreteTestExit(id)) {
+      logger.debug("Get ConcreteTest by id ");
+
```
Comment (Keerthana65) [inline tokyo-supermix-server/src/main/java/com/tokyo/supermix/server/controller/ConcreteTestController.java]: Removed line space & add google style

File: tokyo-supermix-server/src/main/java/com/tokyo/supermix/server/controller/ConcreteTestController.java [hunk 2]
```diff
@@ -1,10 +1,103 @@
...
+      ConcreteTest concreteTest = concreteTestService.getConcreteTestById(id);
+      return new ResponseEntity<>(
+          new ContentResponse<>(Constants.CONCRETE_TEST,
+              mapper.map(concreteTest, ConcreteTestResponseDto.class), RestApiResponseStatus.OK),
```
Comment (Keerthana65) [inline tokyo-supermix-server/src/main/java/com/tokyo/supermix/server/controller/ConcreteTestController.java]: avoid the creating object, here you can directly pass.
 for example mapper.map(concreteTestService.getConcreteTestById(id), ConcreteTestResponseDto.class),

File: tokyo-supermix-server/src/main/java/com/tokyo/supermix/util/Constants.java [hunk 3]
```diff
@@ -234,6 +234,14 @@
   public static final String MIXDESIGNS = "mixdesigns";
   public static final String MIXDESIGN_DELETED = "mix design Successfully Deleted";
   public static final String MIXDESIGN_ID = "mixdesignId";
+  /*
+   *concreteTest*
+   */
+  public static final String ADD_CONCRETE_TEST_SUCCESS = "ConcreteTest added Successfully";
+  public static final String CONCRETE_TEST_DELETED = "ConcreteTest Successfully Deleted";
+  public static final String UPDATE_CONCRETE_TEST_SUCCESS = "ConcreteTest updated Successfully";
+  public static final String CONCRETE_TEST= "concretetest";
```
Comment (Keerthana65) [inline tokyo-supermix-server/src/main/java/com/tokyo/supermix/util/Constants.java]: Used camel case: concreteTest

File: tokyo-supermix-server/src/main/java/com/tokyo/supermix/util/ValidationFailureStatusCodes.java [hunk 4]
```diff
@@ -1079,5 +1079,15 @@ public void setProjectAlreadyExist(String projectAlreadyExist) {
 
   @Value("${validation.project.alreadyExist}")
   private String projectAlreadyExist;
+  // for concreteTest
+  @Value("${validation.concreteTest.notExists}")
+  private String ConcreteTestNotExist;
```
Comment (Keerthana65) [inline tokyo-supermix-server/src/main/java/com/tokyo/supermix/util/ValidationFailureStatusCodes.java]: Used camel case : concreteTestNotExist

File: tokyo-supermix-server/src/main/java/com/tokyo/supermix/server/services/ConcreteTestServiceImpl.java [hunk 5]
```diff
@@ -1,8 +1,64 @@
...
+        mixDesignProportionService.findByMixDesign(mixDesign);
+    Long quantity = null;
+    Long binderquantity = null;
+    for (MixDesignProportion mixDesignProportion : mixDesignProportionList) {
+      if (mixDesignProportion.getRawMaterial().getName().equalsIgnoreCase("cement")
+          || mixDesignProportion.getRawMaterial().getName().equalsIgnoreCase("Fly Ash")) {
+        quantity = mixDesignProportion.getQuantity();
+        binderquantity = mixDesignProportion.getQuantity();
+      }
+
```
Comment (jananthan93) [inline tokyo-supermix-server/src/main/java/com/tokyo/supermix/server/services/ConcreteTestServiceImpl.java]: check the logic

File: tokyo-supermix-server/src/main/java/com/tokyo/supermix/EndpointURI.java [hunk 6]
```diff
@@ -187,6 +187,15 @@
...
   public static final String GET_MIXDESIGN_BY_ID = MIXDESIGN + CODE;
   public static final String DELETE_MIXDESIGN_BY_ID = MIXDESIGN + CODE;
+  /*
+   * concreteTest
+   * 
+   */
+  public static final String CONCRETE_TEST = BASE_API_PATH + "concrete-test";
+  public static final String CONCRETE_TESTS = BASE_API_PATH + "concrete-tests";
+  public static final String DELETE_CONCRETE_TEST = CONCRETE_TEST + ID;
+  public static final String GET_CONCRETE_TEST_BY_ID = CONCRETE_TEST + ID;
```
Comment (Mathuragini) [inline tokyo-supermix-server/src/main/java/com/tokyo/supermix/EndpointURI.java]: You can use CONCRETE_TEST_BY_ID as common for both get and delete.

File: tokyo-supermix-server/src/main/java/com/tokyo/supermix/server/services/ConcreteTestServiceImpl.java [hunk 7]
```diff
@@ -1,8 +1,65 @@
...
+      }
+      if (mixDesignProportion.getRawMaterial().getName().equalsIgnoreCase("FlyAsh")) {
+        binderquantity = mixDesignProportion.getQuantity();
+      }
+    }
+    concreteTest.setWaterCementRatio(concreteTest.getWaterContent() / quantity.doubleValue());
+    concreteTest.setWaterBinderRatio(
+        concreteTest.getWaterContent() / (binderquantity + quantity.doubleValue()));
+    return ConcreteTestRepository.save(concreteTest);
+  }
```
Comment (Mathuragini) [inline tokyo-supermix-server/src/main/java/com/tokyo/supermix/server/services/ConcreteTestServiceImpl.java]: As i already request, one method must contain one functionality. So you have to create separate methods to find water cement ratio and water binder ratio and call to the parent method- save

Comment (Mathuragini) [inline tokyo-supermix-server/src/main/java/com/tokyo/supermix/server/services/ConcreteTestServiceImpl.java]: Consider the logic and create it in separate method

**Pull request merged**